pkg/cache/cache_opts.go

```diff
@@ -139,22 +140,21 @@ func sanitize(c *CacheOpts) error {
 		}
 	}
 
-	if c.Build.Format == CacheBind || c.Launch.Format == CacheBind {
-		var (
-			resolvedPath string
-			err          error
-		)
-		if c.Build.Format == CacheBind {
-			if resolvedPath, err = filepath.Abs(c.Build.Source); err != nil {
-				return errors.Wrap(err, "resolve absolute path")
-			}
-			c.Build.Source = filepath.Join(resolvedPath, "build-cache")
-		} else {
-			if resolvedPath, err = filepath.Abs(c.Launch.Source); err != nil {
-				return errors.Wrap(err, "resolve absolute path")
-			}
-			c.Launch.Source = filepath.Join(resolvedPath, "launch-cache")
+	var (
+		resolvedPath string
+		err          error
+	)
+	if c.Build.Format == CacheBind {
+		if resolvedPath, err = filepath.Abs(c.Build.Source); err != nil {
+			return errors.Wrap(err, "resolve absolute path")
+		}
+		c.Build.Source = filepath.Join(resolvedPath, "build-cache")
+	}
+	if c.Launch.Format == CacheBind {
+		if resolvedPath, err = filepath.Abs(c.Launch.Source); err != nil {
+			return errors.Wrap(err, "resolve absolute path")
```

Comment:
Totally unrelated change... I can pull it out of this PR if desired. Looking through the code it seems the current implementation unnecessarily skips resolving the launch cache when the build cache is a bind.